Error: TypeError: images is not iterable

**Steps to reproduce:** 
When uploading images to chat

**error message:**
submitPrompt.ts:422 Error: TypeError: images is not iterable
    at calculateImageCostByProvider (imageCalculator.ts:15:22)
    at updateImageCosts (submitPrompt.ts:392:21)
    at submitPrompt (submitPrompt.ts:84:3)
    at async PromptBar.handleSubmitPrompt (+page.svelte:86:28)


Need to add test case to pick up this error then fix it

--- a/src/lib/components/prompt-bar/utils/submitPrompt.ts
+++ b/src/lib/components/prompt-bar/utils/submitPrompt.ts
@@ -386,12 +386,15 @@ function updateChatHistory(
  */
 function updateImageCosts(imageArray: any[], currentChatIndex: number): void {
 	let imageCost = 0;
-	let imageTokens = 0;
 
-	for (const image of imageArray) {
-		const imageCalc = calculateImageCostByProvider(image, get(chosenModel), get(chosenCompany));
-		imageCost += imageCalc.cost;
-		imageTokens += imageCalc.tokens;
+	if (imageArray && imageArray.length > 0) {
+		// Calculate costs once for the entire array instead of per image
+		const imageCalc = calculateImageCostByProvider(
+			imageArray,
+			get(chosenModel),
+			get(chosenCompany)
+		);
+		imageCost = imageCalc.cost;
 	}
 
 	chatHistory.update((history) => {

--- a/src/lib/components/settings/GeneralSettingsPage.svelte
+++ b/src/lib/components/settings/GeneralSettingsPage.svelte
@@ -5,7 +5,6 @@
 	import { signOut } from '@auth/sveltekit/client';
 
 	import MoonIcon from '$lib/components/icons/MoonIcon.svelte';
-	import DollarIcon from '$lib/components/icons/DollarIcon.svelte';
 	import LogOutIcon from '$lib/components/icons/LogOutIcon.svelte';
 	import Switch from '$lib/components/Switch.svelte';
 	import type { UserWithSettings } from '$lib/types/types';

--- a/src/lib/models/cost-calculators/imageCalculator.ts
+++ b/src/lib/models/cost-calculators/imageCalculator.ts
@@ -3,16 +3,23 @@ import { ApiProvider } from '@prisma/client';
 
 /**
  * Calculate image cost based on the provider
+ * @param images An array of images or single image object to calculate cost for
+ * @param model The LLM model being used
+ * @param apiProvider The provider of the API (Anthropic, OpenAI, etc.)
+ * @returns Object containing cost and token count
  */
 export function calculateImageCostByProvider(
-	images: Image[],
+	images: Image[] | Image,
 	model: Model,
 	apiProvider: ApiProvider
 ): { cost: number; tokens: number } {
 	let cost = 0;
 	let tokens = 0;
 
-	for (const image of images) {
+	// Handle both array and single image inputs
+	const imageArray = Array.isArray(images) ? images : [images];
+
+	for (const image of imageArray) {
 		let result;
 		if (apiProvider === ApiProvider.anthropic) {
 			result = calculateClaudeImageCost(image.width, image.height, model);

--- a/src/tests/unit/models/cost-calculators/imageCalculator.test.ts
+++ b/src/tests/unit/models/cost-calculators/imageCalculator.test.ts
@@ -73,6 +73,34 @@ describe('Image Cost Calculator', () => {
 			expect(result.cost).toBeCloseTo(image1Result.price + image2Result.price);
 			expect(result.tokens).toBe(image1Result.tokens + image2Result.tokens);
 		});
+
+		it('should handle single image object correctly (non-array input)', () => {
+			// This test checks that the function properly handles when a single image is passed directly
+			// instead of in an array, which might happen from incorrectly formatted data
+			const singleImage = testImages[0];
+
+			// Test the function with a non-array input - should not throw error
+			const result = imageCalculator.calculateImageCostByProvider(
+				singleImage as any, // Cast to any to simulate the bug
+				mockModel,
+				ApiProvider.anthropic
+			);
+
+			// The function should still return a valid result
+			expect(result).toHaveProperty('cost');
+			expect(result).toHaveProperty('tokens');
+			expect(result.cost).toBeGreaterThan(0);
+			expect(result.tokens).toBeGreaterThan(0);
+
+			// Should match the cost of that single image
+			const expectedResult = imageCalculator.calculateClaudeImageCost(
+				singleImage.width,
+				singleImage.height,
+				mockModel
+			);
+			expect(result.cost).toBeCloseTo(expectedResult.price);
+			expect(result.tokens).toBe(expectedResult.tokens);
+		});
 	});
 
 	describe('calculateGptVisionPricing', () => {

--- a/src/tests/unit/prompt-bar/utils/submitPrompt.test.ts
+++ b/src/tests/unit/prompt-bar/utils/submitPrompt.test.ts
@@ -255,9 +255,9 @@ describe('submitPrompt', () => {
 		expect(userMessage?.attachments).toBeDefined();
 		expect(userMessage?.attachments?.length).toBe(1);
 
-		// Verify the calculator was called with the image
+		// Verify the calculator was called with the image array instead of a single image
 		expect(mockCalculateImageCostByProvider).toHaveBeenCalledWith(
-			expect.objectContaining({ url: 'data:image/jpeg;base64,abc123' }),
+			imageArray,
 			expect.anything(),
 			expect.anything()
 		);

